pkg/spi-shared/secretstorage/secretstorage.go

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secretstorage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+// SecretID is a generic identifier of the secret that we store data of. While it very
+// much resembles the Kubernetes client's ObjectKey, we keep it as a separate struct to
```

Comment:
`resembles the Kubernetes client's ObjectKey` this is true now, but I think the biggest benefit is to have option to extend it without changing whole `SecretStorage` interface.